lib/changelog_output.rb

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) 2013-2015 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of version 3 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE about this file by physical or electronic mail,
+# you may find current contact information at www.suse.com
+
+require_relative './changelog_source'
+
+class ChangelogOutput
+
+  def initialize(files)
+    @rpm_changes = files[:rpm_changes]
+    @news = files[:news]
+  end
+
+  def generate
+    changelog = ""
+
+    rpm_changes = ChangelogSource.new(@rpm_changes)
+    rpm_changes.parse
+
+    news = ChangelogSource.new(@news)
+    news.parse
+
+    rpm_changes.elements.each do |version, element|
+      changelog += create_rpm_header(version.to_s, element[:date], element[:email])
+
+
+      element[:items].each_with_index do |item, idx|
+        changelog += "- #{item}\n"
+        news.elements[version][:items].each do |news_item|
+          changelog += "  * #{news_item.gsub(/\n/, "\n  ")}\n" if idx == 0
+        end
+      end
+      changelog += "\n"
+    end
+    changelog
+  end
+
+  def create_rpm_header(version, time, mail)
```

Comment:
Unused method argument - `version`. If it's necessary, use `_` or `_version` as an argument name to indicate that it won't be used.